eclair-core/src/main/scala/fr/acinq/eclair/io/PeerEvents.scala

```diff
@@ -18,9 +18,12 @@ package fr.acinq.eclair.io
 
 import akka.actor.ActorRef
 import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.wire.UnknownMessage
 
 sealed trait PeerEvent
 
-case class PeerConnected(peer: ActorRef, nodeId: PublicKey) extends PeerEvent
+case class PeerConnected(peer: ActorRef, nodeId: PublicKey, data: Peer.Data) extends PeerEvent
```

Comment:
Can you explain why you need to expose the `Peer.Data`? It's internal data, I'd prefer to keep it hidden if possible (and maybe expose only what you found useful for plugins as a new `case class`).

Reply:
To be able to send an `OutgoingMessage` back to `Peer`, a `peerConnection` is needed for that as far as I understand and `Peer.Data` always has the latest one, I use it like this: https://github.com/btcontract/eclair-plugin-messaging-sample/blob/master/src/main/scala/fr/acinq/messagingsample/app/Notifier.scala#L20

Reply:
Also `Peer.Data` has a `remoteInit` which allows a plugin to check if connected remote peer supports this plugin and push some data to remote if it does.

Reply:
Got it, in that case I'd rather only expose what's needed:
```
case class ConnectionInfo(peerConnection: ActorRef, remoteInit: wire.Init)
case class PeerConnected(peer: ActorRef, nodeId: PublicKey, connectionInfo: ConnectionInfo) extends PeerEvent
case class UnknownMessageReceived(peer: ActorRef, nodeId: PublicKey, message: UnknownMessage, , connectionInfo: ConnectionInfo) extends PeerEvent
```